### PR TITLE
Add quota and service IDL proxy configurations.

### PR DESCRIPTION
**Pull request by qiwzhang**
Merge following changes from api-mgmt-poc branch to master:
* attributes.proto: to pass attributes in Proxy config.
* quota.proto:  to generate quota requests in Proxy.
* service_idl.proto:  to define service IDL to extract api attributes

### CODE REVIEW

File: proxy/v1/config/service_idl.proto [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+//              api.operation: istio.mixer.v1.Check
+//         - patterns:
+//            http_method: POST
+//            path: /istio.mixer.v1/Report
+//            attributes:
+//              api.operation: istio.mixer.v1.Report
+//
+//
+// Specifies service IDL with service destination.
+message ServiceIDLSpec {
```
Comment (kyessenov) [inline proxy/v1/config/service_idl.proto]: This doesn't fit well with existing `DestinationPolicy` and `RouteRule`. Maybe add it to destination policy instead?

File: proxy/v1/config/quota.proto [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+  // The final result is "OR" of all match results.
+  repeated MatchClause match = 1;
+
+  // The list of quotas to charge.
+  repeated Quota quotas = 2;
+}
+
+// Specifies a match clause with multiple conditions.
+// The final result is "AND" of all condition results.
+message MatchClause {
```
Comment (kyessenov) [inline proxy/v1/config/quota.proto]: rename to `AttributeMatch`?

**Comment by kyessenov** [comment]
Please rename the PR to be about introducing quota and IDL configurations.

**Comment by qiwzhang** [comment]
It will be split into smaller PR

**Pull request closed without merging**